test/test-str.lisp

```diff
@@ -56,6 +57,7 @@
   (is '("fooxbar" "x") (split "xx" "fooxbarxxx") "split with xx, end in xxx")
   (is '("foo" "bar") (split "(*)" "foo(*)bar"))
   (is '("foo" "bar") (split "NO" "fooNObarNO") "separator at the end (cl-ppcre doesn't return an empty string).")
+  (is '("foo" "bar" "") (split "NO" "fooNObarNO" :limit 10) "but cl-ppcre does return trailing empty strings if limit is provided")
```

Comment:
so this behavior is really unintuitive. We could fix it if we default :limit to (length str)